### DISCUSSION TITLE
Why is verbosity needed in version 8 to make `echo` or `printf` output anything?

### DISCUSSION THREAD

**Discussion by p3k** [google/zx#934] [General] [answered]
Not only because this was different in version 7, but also because I expect when I invoke a command with `$` that it outputs the result of this command to the terminal.

Even the example in the documentation about [`ProcessOutput`](https://google.github.io/zx/getting-started#processoutput) does not output anything in version 8 – except I prepend a line with `$.verbose = true;` (and get more output than I actually want…)

What am I missing?

**Comment by antonmedv** [comment]
>  get more output than I actually want…

Could you please explain? 

ping @antongolub

**Reply by p3k** [reply]
Apologies, I was wrong with claiming that – I did not check again if the output of the invoked commands (the lines starting with `$`) is the same in version 7.

Anyway, the file below, let’s name it `test.mjs`, produces the following outputs:

```js
#!/usr/bin/env zx

//$.verbose = true;
const date = await $`date`;
await $`echo Current date is ${date}.`;
```

* Version 7:
   ```shell
   $ date
   Wed Nov  6 09:26:47 UTC 2024
   $ echo Current date is $'Wed Nov  6 09:26:47 UTC 2024'.
   Current date is Wed Nov  6 09:26:47 UTC 2024.
   ```
* Version 8:  
   _(no output)_
* Version 8 with _uncommented_ `$.verbose = true`:  
   ```shell
   $ date
   Wed Nov  6 09:45:39 UTC 2024
   $ echo Current date is $'Wed Nov  6 09:45:39 UTC 2024'.
   Current date is Wed Nov  6 09:45:39 UTC 2024.
   ```

So I get that with version 8 I would need to enable `$.verbose = true` in every script to get the same output as with version 7.

But still, the example I referred to – or take the one with `Promise.all` on the [“Getting Started” page](https://google.github.io/zx/getting-started#overview)  – just remain silent in version 8.

Is it a documentation issue then?

And it looks like ``$`echo` `` is considered a logging command – is the remedy then to use `echo()` or `console.log()` without having to enable `$.verbose = true`?

**Reply by antongolub** [reply, marked as answer]
I'm afraid, we cannot know in advance how to handle cmd output correctly (whether it contains sensitive data or credentials), so we suggest specifying this behavior explicitly. That's why we use `verbose=false` by default.

**Reply by p3k** [reply]
Thanks for the reply @antongolub – I understand the reason, makes a lot of sense.

I recommend to modify the examples, though – in a way that applying them does not lead to frustration because they yield nothing in the terminal.

Closing this discussion.
